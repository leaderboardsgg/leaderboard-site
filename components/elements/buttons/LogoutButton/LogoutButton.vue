<script setup lang="ts">
import BaseButton from 'elements/buttons/BaseButton/BaseButton.vue'
</script>

<template>
  <BaseButton class="logout-button" v-bind="$attrs"> Log Out </BaseButton>
</template>

<style lang="postcss" scoped>
.logout-button {
  @apply mx-1 py-1 hover:bg-gray-100;
}
</style>
