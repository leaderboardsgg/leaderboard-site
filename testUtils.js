import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
