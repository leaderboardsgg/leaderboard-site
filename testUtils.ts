import { createElementBlock, Component, ComponentOptions } from 'vue'
import { MountingOptions } from '@vue/test-utils'
import userEvent from '@testing-library/user-event'
import { render, RenderOptions } from '@testing-library/vue'

export const stubbedRender = (
  Component: Component | ComponentOptions,
  options?: RenderOptions | MountingOptions<any>,
) => {
  return render(Component, {
    global: {
      stubs: {
        NuxtLink: createElementBlock('a'),
      },
    },
    ...options,
  })
}

export * from '@testing-library/vue'
export * from '@nuxt/test-utils'
export * from '@vue/test-utils'
export { userEvent as fireEvent }
