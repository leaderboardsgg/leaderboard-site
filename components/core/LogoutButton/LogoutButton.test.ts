import LogoutButton from './LogoutButton.vue';
import { stubbedRender } from '@/testUtils';

describe('<LogoutButton />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(LogoutButton);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(LogoutButton);

    expect(container.firstChild).toMatchSnapshot();
  });
});
