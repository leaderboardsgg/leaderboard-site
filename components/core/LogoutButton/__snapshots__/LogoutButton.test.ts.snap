// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButton /> renders correctly 1`] = `
<button
  class="base-button logout-button"
>
  
  Log Out

</button>
`;
